### PR TITLE
HDDS-11642. MutableQuantiles should be stopped

### PR DESCRIPTION
**Pull request by adoroszlai**
## What changes were proposed in this pull request?

`MutableQuantiles` should be stopped when no longer needed, since its `scheduledTask` is referenced by a global `ScheduledExecutorService`.

https://issues.apache.org/jira/browse/HDDS-11642

## How was this patch tested?

Implemented a new Freon test (~not part of the PR~) that keeps creating and closing `RpcClient`:

```
ozone freon occ -t 1 --duration 10m
```

Confirmed on `master` that `MutableQuantiles` instance count is growing:

```
 num     #instances         #bytes  class name (module)
   3:        677544       32522112  org.apache.hadoop.metrics2.lib.MutableQuantiles
   3:        935136       44886528  org.apache.hadoop.metrics2.lib.MutableQuantiles
   3:       1095792       52598016  org.apache.hadoop.metrics2.lib.MutableQuantiles
```

while with the fix it is stable:

```
 num     #instances         #bytes  class name (module)
 319:            18            864  org.apache.hadoop.metrics2.lib.MutableQuantiles
 285:            24           1152  org.apache.hadoop.metrics2.lib.MutableQuantiles
 <no instance>
```

```
pid=$(jps | grep Freon | cut -f1 -d' ')
for i in 1 2 3 4; do sleep 10; jmap -histo:live,file=histogram${i}.txt $pid; done
grep MutableQuantiles histogram*txt
```

CI:
https://github.com/adoroszlai/ozone/actions/runs/11669242466

### CODE REVIEW

File: hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/grpc/metrics/GrpcMetrics.java [hunk 1]
```diff
@@ -100,6 +101,8 @@ public static synchronized GrpcMetrics create(Configuration conf) {
    */
   public void unRegister() {
     DefaultMetricsSystem.instance().unregisterSource(SOURCE_NAME);
+    MetricUtil.stop(grpcProcessingTimeMillisQuantiles);
+    MetricUtil.stop(grpcQueueTimeMillisQuantiles);
```
Comment (jojochuang) [inline hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/grpc/metrics/GrpcMetrics.java:105]:
```suggestion
    MetricUtil.stop(grpcProcessingTimeMillisQuantiles, grpcQueueTimeMillisQuantiles);
```

**Review by jojochuang — APPROVED** [review]
Just a code style suggestion. Apart from that all good.

File: hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/grpc/metrics/GrpcMetrics.java (hunk 1 above)
Comment (adoroszlai) [reply hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/grpc/metrics/GrpcMetrics.java:105]: Thanks @jojochuang for the review.

Both `grpcProcessingTimeMillisQuantiles` and `grpcQueueTimeMillisQuantiles` are arrays, so this does not work.

**Comment by adoroszlai** [comment]
Thanks @jojochuang for the review.

**Pull request merged**